### PR TITLE
feat(routes): deterministic color for hikes

### PR DESCRIPTION
**Pull request by rBergCode**
# Summary
Implemented a deterministic color for each hikes so that the color won't change.

# Details
For this, I get the hash of the id modulo it to get the index of the color to use. The colors are hardcoded and selected by me so that they look good and enough so that there are not too much crossing of hikes of the same color.

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt [hunk 1]
```diff
@@ -130,6 +131,21 @@ data class HikeRoute(
...
+) {
+
+  private val colors =
+      listOf(
+          0xFF7E57C2.toInt(), // Purple
+          0xFF1E88E5.toInt(), // Blue
+          0xFFFFA726.toInt(), // Orange
+          0xFFEF5350.toInt(), // Red
+          0xFF26A69A.toInt() // Teal
+          )
```
Comment (MaelImhof) [inline app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt:143]: Suggestion: could those colors be moved to the `Color.kt` file of the project? Or maybe we have a task for this, I'm not sure.

File: app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt [hunk 2]
```diff
@@ -130,6 +131,21 @@ data class HikeRoute(
...
+
+  /** Get the color of the route from its id. The color should be the same for the same route id. */
+  fun getColor(): Int {
+    return colors[abs(id.hashCode()) % colors.size]
+  }
```
Comment (MaelImhof) [inline app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt:148]: Praise: very nice way of implementing it, and I like the colors you chose as well !

File: app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt [hunk 3]
```diff
@@ -252,8 +254,7 @@ fun HikeDetails(
                       .testTag(TEST_TAG_ELEVATION_GRAPH),
               styleProperties =
                   ElevationGraphStyleProperties(
-                      strokeColor = MaterialTheme.colorScheme.primary,
-                      fillColor = (MaterialTheme.colorScheme.primary).copy(0.1f)))
+                      strokeColor = hikeColor, fillColor = hikeColor.copy(0.1f)))
```
Comment (MaelImhof) [inline app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt:257]: Praise: looks even better now with all the colors !

**Review by MaelImhof — COMMENTED** [review]
Very elegant and nice-looking implementation, the app feels way more "done" now that the colors are actually controlled and systematically the same for a single hike.

Just one comment about the colors, maybe we should put them in `Color.kt`, but maybe that's not even necessary for now, I believe we have a task for the material theme.

File: app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt (hunk 1 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/model/route/HikeRoute.kt:143]: I will do it now

**Review by MaelImhof — APPROVED** [review]
Thanks for the changes, looks perfect to me now !

**Pull request merged**